photon-core/src/main/java/org/photonvision/common/networking/NetworkUtils.java

```diff
@@ -156,15 +157,47 @@ public static NMDeviceInfo getNMinfoForDevName(String devName) {
         return null;
     }
 
+    public static String getActiveConnection(String devName) {
+        var shell = new ShellExec(true, true);
+        try {
+            shell.executeBashCommandNoLog(
+                    "nmcli -g GENERAL.CONNECTION dev show \"" + devName + "\"", true);
+            return shell.getOutput().strip();
+        } catch (Exception e) {
+            logger.error("Exception from nmcli!");
+        }
+        return "";
+    }
+
     public static boolean connDoesNotExist(String connName) {
         var shell = new ShellExec(true, true);
         try {
-            // set nmcli back to DHCP, and re-run dhclient -- this ought to grab a new IP address
-            shell.executeBashCommand("nmcli -f GENERAL.STATE connection show \"" + connName + "\"");
+            shell.executeBashCommandNoLog(
+                    "nmcli -g GENERAL.STATE connection show \"" + connName + "\"", true);
             return (shell.getExitCode() == 10);
         } catch (Exception e) {
             logger.error("Exception from nmcli!");
         }
         return false;
     }
+
+    public static String getIPAddresses(String iFaceName) {
+        List<String> addresses = new ArrayList<String>();
+        try {
+            var iFace = NetworkInterface.getByName(iFaceName);
+            for (var addr : iFace.getInterfaceAddresses()) {
+                var addrStr = addr.getAddress().toString();
+                if (addrStr.startsWith("/")) {
```

Comment:
So this code is to handle this InetBehavior address right?
> The string returned is of the form: hostname / literal IP address

https://docs.oracle.com/javase/8/docs/api/java/net/InetAddress.html#toString--

Reply:
That's correct. If it does return a hostname, I'm not removing it, but if it doesn't, there's no point in leaving the leading "/".